src/LiveComponent/src/Metadata/LiveComponentMetadataFactory.php

```diff
@@ -72,43 +75,54 @@ public function createPropMetadatas(\ReflectionClass $class): array
         return array_values($metadatas);
     }
 
-    public function createLivePropMetadata(string $className, string $propertyName, \ReflectionProperty $property, LiveProp $liveProp): LivePropMetadata
+    public function createLivePropMetadata(string $className, string $propertyName, \ReflectionProperty $property, LiveProp $liveProp): LivePropMetadata|LegacyLivePropMetadata
     {
-        $type = $property->getType();
-        if ($type instanceof \ReflectionUnionType || $type instanceof \ReflectionIntersectionType) {
-            throw new \LogicException(\sprintf('Union or intersection types are not supported for LiveProps. You may want to change the type of property %s in %s.', $property->getName(), $property->getDeclaringClass()->getName()));
-        }
+        // BC layer when "symfony/type-info" is not available
+        if (!method_exists($this->propertyTypeExtractor, 'getType')) {
```

Comment:
Method `$this->propertyTypeExtractor->getTypes()` triggers a deprecation explaining to use `getType()`, which is a bit useless for the end-user since `getTypes()` is internally called by us.

At this point, can we add a deprecation for PHP 8.2+ to tell the user to install `symfony/property-info:^7.1` (which implements `getType()` and ships with `symfony/type-info`)?

Reply:
Hum, I don't see how we can do it correctly, the deprecation on `getTypes()` has been added in `symfony/property-info:^7.3`. 

I guess the current deprecation and a changelog entry could be enough

Reply:
changelog entry added